File: 2580-circular-sentence/circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        stringstream ss(sentence);
        string word;
        ss>>word;
        char f = word[0], ff = word[word.size()-1], n;
        while(ss>>word) {
            n = word[0];
            if(ff!=n) return false;
            ff = word[word.size()-1];
        }

        return f==ff;
    }
};

File: 2580-circular-sentence/circular-sentence.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        String[] words = sentence.split(" ");
        int i;
        for(i=0;i<words.length-1;i++) {
            int n = words[i].length();
            if(words[i].charAt(n-1)!=words[i+1].charAt(0)) return false;
        }
        return words[0].charAt(0)==words[i].charAt(words[i].length()-1);
    }
}